src/node/desktop/src/native/desktop/desktop.cc

```diff
@@ -468,6 +468,67 @@ Napi::Value cleanClipboard(const Napi::CallbackInfo& info)
    return Napi::Value();
 }
 
+Napi::Value shortPathName(const Napi::CallbackInfo& info)
+{
+   Napi::String path = info[0].As<Napi::String>();
+
+#ifdef _WIN32
+
+   // Get path as UTF-8 text
+   std::string utf8Path = info[0].As<Napi::String>().Utf8Value();
+
+   // Figure out required size of string
+   std::wstring widePath(1024, 0);
+   {
+      int size = ::MultiByteToWideChar(
+         CP_UTF8, 0,
+         utf8Path.data(), utf8Path.size(),
+         &widePath[0], widePath.size()
+      );
+
+      if (size == 0) {
+         return path;
+      } else {
+         widePath.resize(size);
```

Comment:
Windows allows paths of up to something like 32767 Unicode characters these days (depending on system configuration), so ideally we'd not assume 1024 as the max. Maybe we won't encounter anything in the real world using paths > 1024, but it's not impossible.

Best to first query for the size (via passing 0 for the last argument), then set the necessary size for the buffer and call again. Does add a bit of overhead but I doubt enough to impact our use of it.